### DISCUSSION TITLE
Livewire encountered corrupt data when trying to hydrate [] component -> only ios

### DISCUSSION THREAD

**Discussion by asterism612** [livewire/livewire#3018] [7. Bugs (archived)]
desktop(window, macos chrome) well.
android well.

fire error only ios(iphone's all browser, ipad's all browser, mac safari)...

**Comment by calebporzio** [comment]
Please include code to reproduce this issue

**Comment by bobwurtz** [comment]
I am having the same issue but only in production. Everything was working fine yesterday morning. I pushed no code updates yesterday. I did not redeploy my production Vapor instance. One of my managers was successfully using this specific form all morning and then later that day it stopped working and started returning this error message. I cannot produce the error outside of production. Running the exact same code locally and in staging produces no error. My Livewire instance looks like:

```
    public function mount()
    {
        $this->setOrganization();

        $this->setOptionCategories();

        $this->setRequirements();

        $this->setOptions();

        $this->setTimeSlots();

        $this->key = Str::random(20);
    }

    public function render()
    {
        $categories = $this->findCategoriesSelect();

        $this->option_categories = Category::where([
            ['organization_id', $this->organization_id],
            ['use_case_id', $this->useCaseId],
            ['type', 'option'],
            ['status', true],
            ['deleted', false]
        ])->get();

        return view('livewire.cafe.action.create-item', compact('categories'));
    }
```

The `option_categories` variable was originally in a `hydrate()` function but I removed it from there because the error message said it found corrupt data trying to hydrate. No help.

I've found several other people with the same problem [here](https://stackoverflow.com/questions/63848826/livewire-encountered-corrupt-data-when-trying-to-hydrate-the-component), [here](https://laracasts.com/discuss/channels/livewire/livewire-encountered-corrupt-data-when-trying-to-hydrate-the-invitationsend-component-ensure-that-the-name-id-data-of-the-livewire-component-wasnt-tampered-with-between-requests?page=1#reply=710714), and [here](https://laracasts.com/discuss/channels/livewire/livewire-encountered-corrupt-data-when-trying-to-hydrate-the), but none of the proposed solutions have helped me. Some of them are simply unanswered.

The strangest thing about this error is that it appeared without me changing anything. Does anyone have any ideas what it could be?

Also - I tried adding unique keys to different elements in the view but that didn't help either. For clarity, the page loads without issue but when I click submit on the form the error appears.

Thanks

**Comment by calebporzio** [comment]
That's tough. Can you strip down the example until submitting works in production, then report the stripped down version that hopefully others or you can test and find the exact problem? Otherwise it's almost impossible to help. Thank you!

**Comment by bobwurtz** [comment]
@calebporzio I was able to solve the problem by changing all Collection instances to arrays by following advice from [this](https://github.com/livewire/livewire/issues/354#issuecomment-841429705) thread (scroll all the way down). All Collections had to be removed even if they were not changed after they were initially declared in `mount()`. Following advice from the other issue, it looks like there is a miscommunication that takes place when data is no longer neatly/plainly stored in a table. Unfortunately, I stumbled upon this by accident so I don't exactly know how to recreate it. Would it be helpful for you if I tried to recreate this issue with a simple app and share what I find? Hopefully I could find some time in the next week or so.

Also, Livewire is awesome and has made my life so much better. Thanks for everything

**Reply by keetamhoang** [reply]
Hi, I have still the same error, can you show the code that mean "hanging all Collection instances to arrays"?

**Comment by tjimenez** [comment]
Seems to be related to Collections and updating any property with that type..

**Comment by hhfa1990** [comment]
The error happen when $state->name contains some special characters.

**index.blade.php**

```
@foreach ($states as $state)
    <livewire:admin.workflow.state.edit :state="$state" wire:key="$state->id"/>
@endforeach
```

**Edit.php**
```
class Edit extends Component
{
    public $state;
    public $name = '';
    public $showModal = false;

    public function mount(State $state)
    {
        $this->name = $state->name;
    }


    public function render()
    {
        return view('livewire.admin.workflow.state.edit');
    }

    public function toggleModal()
    {
        $this->showModal = !$this->showModal;
    }
}
```
**edit.blade.php**
```
<div>
    <a href="#!" wire:click="toggleModal" class="text-indigo-600 hover:text-indigo-900">
        edit
    </a>

    <x-jet-dialog-modal wire:model="showModal">
        <x-slot name="title">Edit status</x-slot>

        <x-slot name="content">
            {{ $state }}
        </x-slot>

        <x-slot name="footer">
            
        </x-slot>
    </x-jet-dialog-modal>
</div>

```

**Reply by ghost** [reply]
Does any one could solved this problem. I have beeen searching 2 weeks finding the answer and this happend to me when I have some special characters in my collection :(  (ios problem)

**Comment by tjimenez** [comment]
Never use special characters or collections. Turn collections into arrays.

**Reply by ghost** [reply]
The problem is I am using Spanish lenguaje so I must use accents :/

**Reply by webcoms** [reply]
Same in German. Strangely its not always with al the special characters. Just some time.

**Reply by ghost** [reply]
> Same in German. Strangely its not always with al the special characters. Just some time.

bro a couple of months ago. 
I thought I'd check the text before saving and using preg_replace and works for me.

 $input['company_detail']= preg_replace('/[^\p{Latin} \x{2600}-\x{1FAFF}0-9+.,()\r\n\-]/u','', $input['company_detail']);

I do not if is the best way but works for me.

**Comment by thanhnguyen0904** [comment]
i have the same problem (only ios problem).

**Comment by shakir-mirza** [comment]
i have the same problem (only ios problem).

**Reply by tjimenez** [reply]
I had to go with inertia, a lot of issues like this happened to me and no answers about it.... :(

**Reply by joshhanley** [reply]
@shakir-mirza any chance you could create a small example repo which reproduces the issue with as little code as possible so we can investigate it? During my tests I can't seem to replicate the issue so if you could, that would be much appreciated, so we can look into getting it fixed!

Thanks!

**Reply by tjimenez** [reply]
@joshhanley sure I'll try to post soon as possible the next week.

**Reply by joshhanley** [reply]
@tjimenez thanks, that'd be great! 🙂

**Reply by le-martre** [reply]
@joshhanley After a few trial and errors, I did it ! https://github.com/MartinVandersteen/safari-accents-bug-livewire

Brand new laravel project, stripped everything in welcome.blade.php and just included the "accents" Livewire component, like so :
```html
<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">

        <title>Livewire Safari Accents bugs</title>
        @livewireStyles
    </head>
    <body>
      @livewire('accents')
      @livewireScripts
    </body>
</html>
```

Accents.php :
```php
<?php

namespace App\Http\Livewire;

use Livewire\Component;

class Accents extends Component
{
  public $string;
  public $text;
  public $number;

  public function mount()
  {
    $this->string = 'â';
    $this->number = 1;
  }

  public function render()
  {
    return view('livewire.accents');
  }

  public function addNumber()
  {
    $this->number++;
  }
}
```

accents.blade.php :
```html
<div> 
  <div>{{ $string }}</div>
  <div>{{ $number }}</div>
  <button wire:click="addNumber">Add number</button>
</div>
```

The $number and addNumber variables and function are really just there to make the component refresh ! Hope it helps you finding the cause of all this because it sure is a headache trying to fix it !


Here are some other characters that might be problematic : éêèà’

**Reply by le-martre** [reply]
Should I turn this into an issue now that we have a way of reproducing the error ? Or should it stay as a discussion ?

**Reply by joshhanley** [reply]
@MartinVandersteen thanks for that! Looks like @jonneroelofs has sorted the issue out 🙂

In regards to logging an issue, this discussion is the issue, as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/livewire/livewire/discussions/3147

**Comment by jonneroelofs** [comment]
I created a PR: https://github.com/livewire/livewire/pull/4613

**Reply by le-martre** [reply]
Can confirm this fixed our issue !! THANK YOU

**Reply by frenchqwerty** [reply]
Me too please !

**Comment by jonneroelofs** [comment]
Could someone create the test as requested by JoshHanley in the PR?
I am not really into browser testing and don't actually have the problem with the special characters myself. So it might take a while before it makes it to the top of my priority list :).
